v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,233 @@
+      {
+        url: getFieldValueFromMessage(message, "website")
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    let addrStr;
+    if (typeof address === "string") {
+      addrStr = address;
+    } else {
+      const { street, city, state, postalCode } = address;
+      addrStr =
+        street && city && state && postalCode
+          ? `${street}, ${city}, ${state}, ${postalCode}`
+          : addrStr;
+    }
+    if (typeof addrStr === "string") {
+      kustomerPayload.locations = [
+        {
+          type: "home",
+          address: addrStr
+        }
+      ];
+    }
+  }
+
+  removeUndefinedAndNullValues(kustomerPayload);
+  return kustomerPayload;
+};
+
+// Main process function responsible for building payload for all
+// type of events.
+const responseBuilderSimple = async (message, category, destination) => {
+  let payload = {};
+  let targetUrl;
+  let userExists = false;
+  // In case of identify type of event first extract the anonymousId, userId
+  // and search if any Kustomer is present in destination.
+  // If present update the same kustomer with the given payload.
+  // else create a new kustomer.
+  //-------------------------------------------------------------
+  // Get Kustomer: https://apidocs.kustomer.com/#ff41f372-6144-4c64-9712-662ee5ef1c33
+  // Create Kustomer: https://apidocs.kustomer.com/#07bd1072-4d4b-4875-b526-8369d711e811
+  // Update Kustomer: https://apidocs.kustomer.com/#077d653a-184e-4153-8133-d24b6427c1ae
+  if (message.type.toLowerCase() == EventType.IDENTIFY) {
+    let response;
+    // If email exists we first search Kustomer with email if present then we mark it
+    // for update call.
+    if (getFieldValueFromMessage(message, "email")) {
+      response = await fetchKustomer(
+        `${BASE_ENDPOINT}/v1/customers/email=${getFieldValueFromMessage(
+          message,
+          "email"
+        )}`,
+        destination
+      );
+      // If Kustomer Exists url to user : Update kustomer
+      if (response && response.status === 200 && response.data.data.id) {
+        userExists = true;
+        targetUrl = `${BASE_ENDPOINT}/v1/customers/${response.data.data.id}?replace=false`;
+      }
+    }
+    // If userExists flag is false
+    // If Kustomer has userId we search using userId as externalId if user
+    // is present or not. If yes then we mark it for update.
+    if (!userExists && getFieldValueFromMessage(message, "userIdOnly")) {
+      response = await fetchKustomer(
+        `${BASE_ENDPOINT}/v1/customers/externalId=${getFieldValueFromMessage(
+          message,
+          "userIdOnly"
+        )}`,
+        destination
+      );
+      // If Kustomer Exists url to user : Update kustomer
+      if (response && response.status === 200 && response.data.data.id) {
+        userExists = true;
+        targetUrl = `${BASE_ENDPOINT}/v1/customers/${response.data.data.id}?replace=false`;
+      }
```

Comment:
You can do this checking/operation in the `fetchKustomer` method to get rid of the code-repetion by making the function return the id or `null`. WDYT?

Reply:
Updated the implementation